homeassistant/components/modbus/fan.py

```diff
@@ -0,0 +1,109 @@
+    CONF_HUB,
+    CONF_REGISTER,
+    CONF_REGISTERS,
+    CONF_VERIFY_STATE,
+    DEFAULT_HUB,
+    MODBUS_DOMAIN,
+)
+from .switch import ModbusCoilSwitch, ModbusRegisterSwitch
+
+_LOGGER = logging.getLogger(__name__)
+
+REGISTERS_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): cv.string,
+        vol.Required(CONF_REGISTER): cv.positive_int,
+        vol.Optional(CONF_COMMAND_OFF, default=0): cv.positive_int,
+        vol.Optional(CONF_COMMAND_ON, default=1): cv.positive_int,
+        vol.Optional(CONF_HUB, default=DEFAULT_HUB): cv.string,
+        vol.Optional(CONF_SLAVE): cv.positive_int,
+        vol.Optional(CONF_VERIFY_STATE, default=True): cv.boolean,
+    }
+)
+
+COILS_SCHEMA = vol.Schema(
+    {
+        vol.Required(CALL_TYPE_COIL): cv.positive_int,
+        vol.Required(CONF_NAME): cv.string,
+        vol.Required(CONF_SLAVE): cv.positive_int,
+        vol.Optional(CONF_HUB, default=DEFAULT_HUB): cv.string,
+    }
+)
+
+PLATFORM_SCHEMA = vol.All(
+    cv.has_at_least_one_key(CONF_COILS, CONF_REGISTERS),
+    PLATFORM_SCHEMA.extend(
+        {
+            vol.Optional(CONF_COILS): [COILS_SCHEMA],
+            vol.Optional(CONF_REGISTERS): [REGISTERS_SCHEMA],
+        }
+    ),
+)
+
+
+async def async_setup_platform(hass, config, add_entities, _):
+    """Read configuration and create Modbus fan."""
+    fans = []
+    if CONF_COILS in config:
+        for coil in config[CONF_COILS]:
+            hub_name = coil[CONF_HUB]
+            hub = hass.data[MODBUS_DOMAIN][hub_name]
+            fans.append(
+                ModbusCoilFan(
+                    hub, coil[CONF_NAME], coil[CONF_SLAVE], coil[CALL_TYPE_COIL]
+                )
+            )
+    if CONF_REGISTERS in config:
+        for register in config[CONF_REGISTERS]:
+            hub_name = register[CONF_HUB]
+            hub = hass.data[MODBUS_DOMAIN][hub_name]
+
+            fans.append(
+                ModbusRegisterFan(
+                    hub,
+                    register[CONF_NAME],
+                    register.get(CONF_SLAVE),
+                    register[CONF_REGISTER],
+                    register[CONF_COMMAND_ON],
+                    register[CONF_COMMAND_OFF],
+                    register[CONF_VERIFY_STATE],
+                )
+            )
+
+    add_entities(fans)
+
+
+class ModbusCoilFan(ModbusCoilSwitch, FanEntity):
+    """Representation of a Modbus coil fan."""
+
+
+class ModbusRegisterFan(ModbusRegisterSwitch, FanEntity):
+    """Representation of a Modbus register fan."""
+
+    def __init__(
+        self, hub, name, slave, register, command_on, command_off, verify_state,
+    ):
+        """Initialize the register fan."""
+        super().__init__(
+            hub=hub,
+            name=name,
+            slave=slave,
+            register=register,
+            command_on=command_on,
+            command_off=command_off,
+            verify_state=verify_state,
+            register_type=CALL_TYPE_REGISTER_HOLDING,
+        )
```

Comment:
I cannot find the difference from a switch, it would be nice to have at least one difference :-)

Reply:
@janiversen the main difference is how the entity appears in the UI. Technically, it's still a switch but using a web component for a fan. Same holds for the light entity.